epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggableHandler.java

```diff
@@ -27,11 +34,37 @@ public boolean isLoggable(Element element) {
   @Override
   public String logInvocation(Element element) {
     TypeMirror dataType = dataType(element);
-    var reflectedType =
-        m_processingEnv
-            .getElementUtils()
-            .getTypeElement(m_processingEnv.getTypeUtils().erasure(dataType).toString());
+    TypeElement reflectedType =
+        (TypeElement) m_processingEnv.getTypeUtils().asElement(m_typeUtils.erasure(dataType));
+
+    List<TypeMirror> subtypes = getLoggedSubtypes();
+
+    return subtypes.isEmpty()
+        ? generateSimpleLogInvocation(reflectedType, element)
+        : generateConditionalLogInvocation(reflectedType, element, subtypes);
+  }
+
+  private List<TypeMirror> getLoggedSubtypes() {
+    List<TypeMirror> loggedSubtypes = new ArrayList<>();
+    for (TypeMirror subtype : getAllSubtypes()) {
+      Element subtypeElement = m_typeUtils.asElement(subtype);
+      if (subtypeElement != null && subtypeElement.getAnnotation(Logged.class) != null) {
+        loggedSubtypes.add(subtype);
+      }
+    }
+    return loggedSubtypes;
+  }
+
+  private List<TypeMirror> getAllSubtypes() {
+    // This method should return all known subtypes of the given type.
```

Comment:
What given type? You didn't give this function any parameters (note: you'll need to add some)